honeycombio/resource_marker_test.go

```diff
@@ -31,35 +38,54 @@ func TestAccHoneycombioMarker_basic(t *testing.T) {
 	})
 }
 
-func testAccMarkerConfig(dataset string) string {
-	return fmt.Sprintf(`
-resource "honeycombio_marker" "test" {
-  message = "Hello world!"
-  type    = "deploys"
-  url     = "https://www.honeycomb.io/"
-  dataset = "%s"
-}`, dataset)
+func TestAccHoneycombioMarker_AllToUnset(t *testing.T) {
+	ctx := context.Background()
+	c := testAccClient(t)
+
+	if c.IsClassic(ctx) {
+		t.Skip("env-wide markers are not supported in classic")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 testAccPreCheck(t),
+		ProtoV5ProviderFactories: testAccProtoV5ProviderFactory,
+		Steps: []resource.TestStep{
+			{
+				Config: `
+resource "honeycombio_marker" "test" {		
+  message = "hey"
+	type    = "test"
+  dataset = "__all__"
+}`,
+				Check: testAccCheckMarkerExists(t, "honeycombio_marker.test", honeycombio.EnvironmentWideSlug),
+			},
+			{
+				Config: `
+resource "honeycombio_marker" "test" {		
+  message = "hey"
+	type    = "test"
+}`,
+				Check:              testAccCheckMarkerExists(t, "honeycombio_marker.test", honeycombio.EnvironmentWideSlug),
+				PlanOnly:           true,
+				ExpectNonEmptyPlan: false,
+			},
+		},
+	})
 }
 
-// testAccCheckMarkerExists queries the API to verify the Marker exists and
-// matches with the Terraform state.
-func testAccCheckMarkerExists(t *testing.T, name string) resource.TestCheckFunc {
+func testAccCheckMarkerExists(t *testing.T, name, dataset string) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		resourceState, ok := s.RootModule().Resources[name]
 		if !ok {
 			return fmt.Errorf("not found: %s", name)
 		}
 
 		c := testAccClient(t)
-		m, err := c.Markers.Get(context.Background(), resourceState.Primary.Attributes["dataset"], resourceState.Primary.ID)
+		_, err := c.Markers.Get(context.Background(), dataset, resourceState.Primary.ID)
 		if err != nil {
 			return fmt.Errorf("could not retrieve marker: %w", err)
 		}
 
-		assert.Equal(t, "Hello world!", m.Message)
```

Comment:
why'd these bits get removed?

Reply:
Ah, I should have commented!

They're being asserted above using the native functionality already (so we're doing it twice): this function should just be asserting that the marker exists remotely

Reply:
Got it - thank you!